app/controllers/locations/map_controller.rb

```diff
@@ -0,0 +1,27 @@
+require_dependency "locations/application_controller"
+
+module Locations
+  class MapController < ApplicationController
+
+    def index
+        @map = Map.new(id: map_params[:id],loc: map_params[:loc])
+
+        if !@map.url.nil?
+          if @map.url.include? "http"
+            redirect_to @map.url
+          else
+            render plain: "#{@map.url}", status: 200
```

Comment:
I would return that logic model. Maybe adding something like a .reserve? method that the controller can call to test if it should return a message or the map.url value would be the way to go in terms of handling the conditional at that level. Considering the update to the locator as a whole is going to be a bit down the road is probably better to render this in an actual view so users get a reasonably formatted page.